src/common/ERC1271.sol

```diff
@@ -3,117 +3,35 @@ pragma solidity ^0.8.23;
-    /// - https://github.com/frangio/eip712-wrapper-for-eip1271
-    ///
-    /// Of course, if you are a wallet app maker and can update your app's UI at will,
-    /// you can choose a more minimalistic signature scheme like
-    /// `keccak256(abi.encode(address(this), hash))` instead of all these acrobatics.
-    /// All these are just for widespead out-of-the-box compatibility with other wallet apps.
-    ///
-    /// The `hash` parameter is the `childHash`.
-    function isValidSignature(bytes32 hash, bytes calldata signature) public view virtual returns (bytes4 result) {
-        /// @solidity memory-safe-assembly
-        assembly {
-            let m := mload(0x40) // Cache the free memory pointer.
-            let o := add(signature.offset, sub(signature.length, 0x60))
-            calldatacopy(0x00, o, 0x60) // Copy the `DOMAIN_SEP_B` and child's structHash.
-            mstore(0x00, 0x1901) // Store the "\x19\x01" prefix, overwriting 0x00.
-            for {} 1 {} {
-                // Use the nested EIP-712 workflow if the reconstructed childHash matches,
-                // and the signature is at least 96 bytes long.
-                if iszero(or(xor(keccak256(0x1e, 0x42), hash), lt(signature.length, 0x60))) {
-                    // Truncate the `signature.length` by 3 words (96 bytes).
-                    signature.length := sub(signature.length, 0x60)
-                    mstore(0x00, calldataload(o)) // Store the `PARENT_TYPEHASH`.
-                    mstore(0x20, hash) // Store the `childHash`.
-                    // The child's structHash is already at 0x40.
-                    hash := keccak256(0x00, 0x60) // Compute the parent's structHash.
-                    break
-                }
-                // Else, use the `personal_sign` workflow.
-                // Truncate the `signature.length` by 1 word (32 bytes), until zero.
-                signature.length := mul(gt(signature.length, 0x20), sub(signature.length, 0x20))
-                // The `PARENT_TYPEHASH` is already at 0x40.
-                mstore(0x60, hash) // Store the `childHash`.
-                hash := keccak256(0x40, 0x40) // Compute the parent's structHash.
-                mstore(0x60, 0) // Restore the zero pointer.
-                break
-            }
-            mstore(0x40, m) // Restore the free memory pointer.
-        }
-        bool success = _isValidSignature(_hashTypedData(hash), signature);
-        /// @solidity memory-safe-assembly
-        assembly {
-            // `success ? bytes4(keccak256("isValidSignature(bytes32,bytes)")) : 0xffffffff`.
-            result := shl(224, or(0x1626ba7e, sub(0, iszero(success))))
+    /// @dev bytes4(keccak256("isValidSignature(bytes32,bytes)"))
+    bytes4 internal constant _1271_MAGIC_VALUE_SUCCESS = 0x1626ba7e;
+    bytes4 internal constant _1271_MAGIC_VALUE_FAILURE = 0xffffffff;
+    bytes32 internal constant _MESSAGE_TYPEHASH = keccak256("LightAccountMessage(bytes message)");
```

Comment:
(No change needed) Do you know what the rationale was for the field being a `bytes` instead of a `bytes32`? I might have something a little hazy here, but I believe this field is always containing the hash that's being asked to be signed, right?

Reply:
Thanks for the review!

I was thinking I might want to change that too, but left it as is. This is also what MultiOwnerPlugin has.

I believe the original reason was because that's what Safe does. I'm happy to change it if it won't cause SDK headaches.

Cc @howydev

Reply:
Thanks for the explanation. I think it's fine to stick to what Safe does, I was just curious.

Reply:
Yep, we inherited that conversion of `bytes32 digest -> encoded into bytes -> hashed back to a bytes32` from Safe. It added a little overhead that's probably unnecessary, but I think the SDK already accounts for this. 100% should address this in the next iteration though

Reply:
Cool, thanks guys